### PR TITLE
fix 3 non-matchings in z_demo.c

### PR DESCRIPTION
**Pull request by pixel-stuck**

### CODE REVIEW

**Comment by Random06457** [comment]
It would be better to use the appropriate macros for the button values:
```
-0x1001 : ~START_BUTTON
0xFFFF7FFF : ~A_BUTTON
-0x4001 : ~B_BUTTON
-0x101 : ~R_JPAD
-0x201 : ~L_JPAD
-0x801 : ~U_JPAD
```

File: src/code/z_demo.c [hunk 1]
```diff
@@ -95,19 +95,17 @@ void func_80064558(GlobalContext* globalCtx, CutsceneContext* csCtx) {
     }
 }
 
-#ifdef NON_MATCHING
-// minor ordering and regalloc differences
 void func_800645A0(GlobalContext* globalCtx, CutsceneContext* csCtx) {
-    CutsceneStateHandler handler;
+    Input *player1 = &globalCtx->state.input[0];
```
Comment (Roman971) [inline src/code/z_demo.c:99]: I'm not sure about using the term "player" in this context.
Also pointers should have the star on the left side based on our style.

File: src/code/z_demo.c [hunk 2]
```diff
@@ -195,6 +187,7 @@ void func_80064824(GlobalContext* globalCtx, CutsceneContext* csCtx, CsCmdBase*
             if (sp3F != 0) {
                 func_800F6D58(0xF, 0, 0);
                 func_800753C4(globalCtx, 3);
+                if(1);
```
Comment (Roman971) [inline src/code/z_demo.c:190]: This doesn't fit our style as is.
```suggestion
                if (1) {}
```

File: src/code/z_demo.c (hunk 1 above)
Comment (pixel-stuck) [reply src/code/z_demo.c]: renamed to "pad1" and corrected the star.

File: src/code/z_demo.c (hunk 2 above)
Comment (pixel-stuck) [reply src/code/z_demo.c]: fixed.

File: src/code/z_demo.c (hunk 1 above)
Comment (fig02) [reply src/code/z_demo.c]: pad1 makes sense but also might be confused for stack padding (lmao). Maybe controller1?

Comment (pixel-stuck) [reply src/code/z_demo.c]: I can do that, I mostly named it that because `pad1->padPressed`

**Pull request merged**